pkgs/development/python-modules/greatfet/default.nix

```diff
@@ -1,34 +1,54 @@
 {
-  lib,
-  fetchFromGitHub,
   buildPythonPackage,
-  isPy3k,
+  cmsis-svd,
+  fetchFromGitHub,
   future,
-  pyusb,
   ipython,
+  lib,
+  prompt-toolkit,
+  pyfwup,
   pygreat,
+  pythonOlder,
+  pyusb,
+  setuptools-git-versioning,
+  setuptools,
+  tabulate,
+  tqdm,
 }:
 
 buildPythonPackage rec {
   pname = "greatfet";
   version = "2024.0.1";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "greatscottgadgets";
     repo = "greatfet";
     rev = "refs/tags/v${version}";
-    sha256 = "sha256-AKpaJZJTzMY3IQXLvVnLWh3IHeGp759z6tvaBl28BHQ=";
+    hash = "sha256-AKpaJZJTzMY3IQXLvVnLWh3IHeGp759z6tvaBl28BHQ=";
   };
 
-  disabled = !isPy3k;
+  disabled = pythonOlder "3.8";
+
+  build-system = [
+    setuptools
+    setuptools-git-versioning
+  ];
 
-  propagatedBuildInputs = [
+  dependencies = [
+    cmsis-svd
     future
-    pyusb
     ipython
+    prompt-toolkit
+    pyfwup
     pygreat
+    pyusb
+    tabulate
+    tqdm
   ];
 
+  # Tests seem to require devices (or simulators) which are
+  # not available in the build sandbox.
   doCheck = false;
 
   preBuild = ''
```

Comment:
```suggestion
  postPatch = ''
```

Reply:
You could also use `sourceRoot`.

Reply:
I don't think the Python builder supports that?

Reply:
That disables access to `..`, which we need to write the version

Reply:
No it doesn't.

Reply:
Shouldn't matter now, as of the test

But when trying to access `..` with `sourceRoot` set, it just pointed into `/nix/store`

Reply:
It does point to the correct directory but only with read access.

Reply:
Ok, maybe I made a mistake somewhere, not sure.
But as I said, it shouldn't matter now